libs/ha/ha_common_libs_k8s.py

```diff
@@ -2038,3 +2039,79 @@ def create_bkt_put_object(self, event, s3_test, bkt, obj, **kwargs):
+        :param background: Flag to start background process
+        :return: (bool, dict)
+        """
+        checksums = dict()
+        fail_count = 0
+        exp_fail_count = 0
+        test_dir_path = os.path.join(TEST_DATA_FOLDER, "HAOverWrite")
+        if not os.path.isdir(test_dir_path):
+            LOGGER.debug("Creating file path %s", test_dir_path)
+            system_utils.make_dirs(test_dir_path)
+
+        resp = s3_test_obj.bucket_list()[1]
+        for bkt, value in s3_data.items():
+            object_name = value[0]
+            object_size = value[1]
+            if bkt not in resp:
+                LOGGER.info("Creating a bucket with name %s and uploading object of size %s MB",
+                            bkt, object_size)
+                file_path = os.path.join(test_dir_path, f"{bkt}.txt")
+                _ = s3_test_obj.create_bucket_put_object(bkt, object_name, file_path, object_size)
+                LOGGER.info("Created a bucket with name %s and uploaded object %s of size %s MB",
+                            bkt, object_name, object_size)
+                system_utils.remove_file(file_path)
+
+        LOGGER.info("Total Iteration : %s", iteration)
+        for bucket_name, value in s3_data.items():
+            object_name = value[0]
+            object_size = value[1]
+            LOGGER.info("Bucket Name : %s", bucket_name)
+            LOGGER.info("Object Name : %s", object_name)
+            LOGGER.info("Max Object size : %s MB", object_size)
+            for i_i in range(iteration):
+                loop = i_i + 1
+                LOGGER.info("Iteration : %s", loop)
+                file_size = random.SystemRandom().randint(0, object_size) if random_size \
+                    else object_size
+                try:
+                    upload_path = os.path.join(test_dir_path, f"{object_name}_upload.txt")
+                    LOGGER.info("Creating a file with name %s", object_name)
+                    system_utils.create_file(upload_path, file_size, "/dev/urandom", '1M')
+                    LOGGER.info("Retrieving checksum of file %s", upload_path)
+                    up_checksum = self.cal_compare_checksum([upload_path], compare=False)[0]
+                    LOGGER.info("Uploading object (Overwriting)...")
+                    _ = s3_test_obj.put_object(bucket_name, object_name, upload_path)
+
+                    LOGGER.info("Downloading object...")
+                    download_path = os.path.join(test_dir_path, f"{object_name}_download.txt")
+                    _ = s3_test_obj.object_download(bucket_name, object_name, download_path)
+                    dnld_checksum = self.cal_compare_checksum([download_path], compare=False)[0]
+                    checksums[f"{bucket_name}_{loop}"] = [up_checksum, dnld_checksum]
+                except CTException as error:
+                    if event.is_set:
+                        LOGGER.error("Event is set, overwrite/object download failure is expected. "
```

Comment:
do we want to print iteration as well in case of any failure so we will know in test?

Reply:
We are printing it on line 2087.